interop-testing/src/test/java/io/grpc/testing/integration/NettyFlowControlTest.java

```diff
@@ -0,0 +1,374 @@
+
+    // deal with cases that either don't cause a window update or hit max window
+    expectedWindow = Math.min(MAX_WINDOW, (Math.max(expectedWindow, REGULAR_WINDOW)));
+
+    // I know this range looks large, but this allows for only one extra/missed window update
+    // (one extra update causes a 2x difference and one missed update causes a .5x difference)
+    if ((lastWindow < 2.1 * expectedWindow) && (expectedWindow < 2.1 * lastWindow)) {
+      System.out.println("PASS");
+      testsPassed++;
+    } else {
+      System.out.println("FAIL");
+      testsFailed++;
+    }
+  }
+
+  /**
+   * Resets client/server and their flow control windows
+   *
+   * @param serverFlowControlWindow
+   * @param clientFlowControlWindow
+   */
+  private void resetConnection(int serverFlowControlWindow, int clientFlowControlWindow) {
+    if (channel != null) {
+      if (!channel.isShutdown()) {
+        channel.shutdown();
+      }
+    }
+    startServer(serverFlowControlWindow);
+
+    int channelPort = port;
+    if (useProxy) {
+      channelPort = proxyPort;
+    }
+    channel = NettyChannelBuilder.forAddress(new InetSocketAddress(loopback, channelPort))
+        .flowControlWindow(clientFlowControlWindow).negotiationType(NegotiationType.PLAINTEXT)
+        .build();
+  }
+
+  private void startServer(int serverFlowControlWindow) {
+    ServerBuilder<?> builder = NettyServerBuilder.forAddress(new InetSocketAddress(loopback, port))
+        .flowControlWindow(serverFlowControlWindow);
+    List<ServerInterceptor> allInterceptors = ImmutableList.<ServerInterceptor>builder()
+        .build();
+    builder.addService(ServerInterceptors.intercept(
+        TestServiceGrpc.bindService(new TestServiceImpl(Executors.newScheduledThreadPool(2))),
+        allInterceptors));
+    try {
+      server = builder.build().start();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  /**
+   * Restart proxy with new bandwidth/latency settings.
+   *
+   * @param targetBPS
+   * @param targetLatency
+   * @param latencyUnits
+   */
+  private void resetProxy(final int targetBPS, final int targetLatency,
+      final TimeUnit latencyUnits) {
+    executor.submit(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          proxy.reset(targetBPS, targetLatency, latencyUnits);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+    });
+    try {
+      // give the a few millis to start before trying to re-connect. Without this, the first request
+      // sometimes throws an HTTP/2 error for a bad message.
+      Thread.sleep(10);
+    } catch (InterruptedException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  private static void startProxy() {
+    executor.submit(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          proxy.start();
+        } catch (Exception e) {
+          throw new RuntimeException(e);
```

Comment:
This will probably get lost, since the executor may ignore the exception.  You should return the Future from `execute` from startproxy, and somehow wait on it (perhaps in test setup)